[4llysa] iP

4llysa opened the pull request:
**DukePro**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ))


DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

***
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

4llysa committed:
Update Duke.java

4llysa committed:
Update Duke.java

4llysa committed:
Update Duke.java

4llysa committed:
Update Duke.java

4llysa committed:
Update Duke.java and Task inheritance

4llysa committed:
Remove unused variables in Todo.java

4llysa committed:
Update EXPECTED.TXT

4llysa committed:
Add AllyException.java

4llysa committed:
Update Duke.java

4llysa committed:
Update Duke.java

4llysa committed:
Update Duke.java

4llysa committed:
Update Duke.java

4llysa committed:
Update Duke.java and Task.java

Task now implements Serializable interface

4llysa committed:
Update Duke.java

4llysa committed:
Update .gitignore

Ignore data/duke.txt

4llysa committed:
Update Duke.java and Task.java

4llysa committed:
Update Deadline.java, Duke.java, Event.java

Update Level 8 increments using LocalDate

4llysa committed:
Add TaskList.java

Add TaskList Class to improve OOP

4llysa committed:
Add Parser.java

Add Parser Class to improve OOP

4llysa committed:
Add Storage.java

Add Storage Class to improve OOP

4llysa committed:
Update TaskList.java

Duke does not fully use TaskList as an abstraction

The TaskList should be used for abstraction of addition and deletion of tasks

4llysa committed:
Add UI.java

4llysa committed:
Update Duke.java

4llysa committed:
Add Package Directories

4llysa committed:
Merge remote-tracking branch 'origin/add-gradle-support'

4llysa committed:
Add Gradle Support and Tests

4llysa committed:
Update Storage.java

Previous version does not support creation of Duke.txt file if not found

4llysa committed:
Update Javadocs class descriptions

4llysa committed:
Update TaskListTest.java

4llysa committed:
Add Finder.java

4llysa committed:
Update import statements

4llysa committed:
Update javadocs for all non-private methods

4llysa committed:
Update javadocs for all non-private methods

4llysa committed:
Add Search functionality

4llysa committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java

4llysa committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java

4llysa committed:
Add checkstyle test support

4llysa committed:
Attempt to use JavaFX

Work in progress

Still need to integrate fxml

ongzhili commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,208 @@
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/user.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/duke.png"));
+
+    protected static TaskList lst = new TaskList();
```
Since TaskList extends ArrayList<Task>, would it be more appropriate to use a multi-valued variable name?

ongzhili commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,208 @@
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+import duke.ui.UI;
```
Would it be good here to leave a line to separate the different import packages?

ongzhili commented on src/main/java/duke/Finder.java:
```diff
@@ -0,0 +1,41 @@
+     * @return boolean found flag
+     */
+    void find(String s) {
+        boolean found  = false;
```
Could this be named to sound more like a boolean? I suggest isFound as a possible alternative.

ongzhili commented on src/main/java/duke/ui/UI.java:
```diff
@@ -0,0 +1,50 @@
+            System.out.println(ctr + ": " + t);
+            ctr++;
+        }
+        if (lst.size()==0) {
```
Would it be slightly clearer if the '==' is surrounded by whitespace?
```suggestion
        if (lst.size() == 0) {
```

ongzhili commented on src/test/java/duke/StorageTest.java:
```diff
@@ -0,0 +1,71 @@
+
+/**
+ * Test for Storage class used in Duke.
+ */
```
I like that each test class only tests one thing at a time, in case one test fails, the rest of the tests can still run.

ongzhili reviewed (commented):
Seems good 👍 

Some minor coding standard and code naming suggestions that you can consider.

4llysa committed:
Add Main.java and MainWindow.java controller

This new version now supports JavaFXML

4llysa committed:
Update Duke to support the same functions as CLI

4llysa committed:
Update Duke use Varargs when necessary

4llysa committed:
Update Duke to exit gracefully

4llysa committed:
Update build.gradle

build.gradle does not support shadowJar

Jar files are useful for distribution of the Duke Chat Bot

Let's utilise the shadowJar plugin in gradle to create a jar file

4llysa committed:
Add assertions

4llysa committed:
Update TaskList.java to follow Java code styles

4llysa committed:
Update *.java to follow Java code styles

4llysa committed:
Update Duke.java, TaskListTest.java, UI.java to follow Java code styles

4llysa commented:
Thanks for the feedback!

4llysa committed:
Duke.java: Update to work with GUI
UI.java: Remove UI.java as the CLI is now necessary

4llysa committed:
Merge pull request #1 from 4llysa/branch-A-Assertions

Add Assertions

4llysa committed:
Merge remote-tracking branch 'origin/master'

4llysa committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/ui/UI.java

4llysa committed:
Add Archive Function to Duke

4llysa committed:
Rename to Ally

4llysa committed:
Improve Code Quality

4llysa committed:
Update user docs

4llysa committed:
Rename Ui.png

4llysa committed:
Add img.png

4llysa committed:
Update User Guide

4llysa committed:
Update User Guide

4llysa committed:
Rename Jar File

4llysa committed:
Improve Code Quality and Standards

4llysa committed:
Improve Code Quality and Standards

4llysa committed:
Update User Guide

4llysa committed:
Update JUnit Tests

4llysa committed:
Update JUnit test

4llysa committed:
Update Code Quality

4llysa committed:
Update Code based on CheckStyle

4llysa committed:
Add Title of Window in Main Window

4llysa committed:
Update User Guide

4llysa committed:
Update TaskList to handle bugs arising from user input error

4llysa committed:
Update User Docs

4llysa committed:
Update User Docs

4llysa committed:
Update User Docs

4llysa committed:
Update TaskList.java for minor typo

4llysa committed:
Update TaskList.java to fix crashing bug from incorrect user input

4llysa committed:
Update JUnit Tests to be more comprehensive
